### PR TITLE
Update java.md

### PR DESCRIPTION
**Pull request by z4kn4fein**
### Description

Docs related follow up changes from [this](https://github.com/configcat/java-sdk/pull/10) PR.

### Requirement checklist

- [x] I have validated my changes on a test/local environment.
- [ ] I have checked the SNYK/Dependabot reports and applied the suggested changes.
- [ ] (Optional) I have updated outdated packages.

### CODE REVIEW

File: website/docs/sdk-reference/java.md [hunk 1]
```diff
@@ -154,40 +154,48 @@ java.util.Collection<String> keys = client.getAllKeys();
 ## Polling Modes
 The *ConfigCat SDK* supports 3 different polling mechanisms to acquire the setting values from *ConfigCat*. After latest setting values are downloaded, they are stored in the internal cache then all `getValue()` calls are served from there. With the following polling modes, you can customize the SDK to best fit to your application's lifecycle.
 
-### Auto polling (default)
+### Auto Polling (default)
```
Comment (laliconfigcat) [inline website/docs/sdk-reference/java.md:157]:
```suggestion
### Auto polling (default)
```

File: website/docs/sdk-reference/java.md [hunk 2]
```diff
@@ -47,7 +47,7 @@ You can safely shut down the client instance and release all associated resource
 client.close()
 ```
 
-### Java compatibility
+## Java Compatibility
```
Comment (laliconfigcat) [inline website/docs/sdk-reference/java.md:50]:
```suggestion
## Java compatibility
```

File: website/docs/sdk-reference/java.md [hunk 3]
```diff
@@ -154,40 +154,48 @@ java.util.Collection<String> keys = client.getAllKeys();
...
     .build("<PLACE-YOUR-SDK-KEY-HERE>");
 ```
 
-### Lazy loading
+Available options:
+
+| Option Parameter                    | Description                              | Default |
+| ----------------------------------- | ---------------------------------------- | ------- |
+| `autoPollIntervalInSeconds`         | Polling interval.                        | 60      |
+| `maxInitWaitTimeSeconds`            | Maximum waiting time between the client initialization and the first config acquisition in secconds. | 5       |
```
Comment (laliconfigcat) [inline website/docs/sdk-reference/java.md:183]:
```suggestion
| `maxInitWaitTimeSeconds`            | Maximum waiting time between the client initialization and the first config acquisition in seconds. | 5       |
```

File: website/docs/sdk-reference/java.md [hunk 4]
```diff
@@ -196,18 +204,139 @@ ConfigCatClient client = ConfigCatClient.newBuilder()
 ```
 If you set the `asyncRefresh` to `false`, the refresh operation will be awaited until the fetching of the new configuration is completed.
 
-### Manual polling
+Available options:
+
+| Option Parameter                | Description                  | Default |
+| ------------------------------- | ---------------------------- | ------- |
+| `cacheRefreshIntervalInSeconds` | Cache TTL.                   | 60      |
+| `useAsyncRefresh`               | Asynchronously refresh.      | false   |
```
Comment (laliconfigcat) [inline website/docs/sdk-reference/java.md:212]:
```suggestion
| `useAsyncRefresh`               | Asynchronous refresh.      | false   |
```

File: website/docs/sdk-reference/java.md [hunk 5]
```diff
@@ -196,18 +204,139 @@ ConfigCatClient client = ConfigCatClient.newBuilder()
...
+        "local_flags.json", // name of the resource
+        true // reload the resource when it gets modified
+    ))
+    .build("localhost");
+```
+#### JSON File Structure
+The SDK supports 2 types of JSON structures to describe feature flags & settings.
+
+The first one is in the same format that the SDK fetches from the ConfigCat CDN. This format allows the usage of all features you can do on the ConfigCat Dashboard.
+```json
```
Comment (laliconfigcat) [inline website/docs/sdk-reference/java.md:253]: Ezeket én lehet nem erőltetném ide, hanem csinálnék egy sample_complex meg sample_simple .json-t, és csak linkelnék rá. I don't know, dumáljunk róla.

File: website/docs/sdk-reference/java.md [hunk 6]
```diff
@@ -196,18 +204,139 @@ ConfigCatClient client = ConfigCatClient.newBuilder()
...
+{
+    "f": { // list of feature flags & settings
+        "isFeatureEnabled": { // key of a particular flag
+            "v": false, // default value, served when no rules are defined
+            "i": "430bded3", // variation id (for analytical purposes)
+            "t": 0, // flag's type, possible values:
```
Comment (laliconfigcat) [inline website/docs/sdk-reference/java.md:259]:
```suggestion
            "t": 0, // featere flag's type, possible values: 
```

File: website/docs/sdk-reference/java.md [hunk 7]
```diff
@@ -196,18 +204,139 @@ ConfigCatClient client = ConfigCatClient.newBuilder()
...
+                        // 10 -> '= (Number)',
+                        // 11 -> '<> (Number)',
+                        // 12 -> '< (Number)',
+                        // 13 -> '<= (Number)',
+                        // 14 -> '> (Number)',
+                        // 15 -> '>= (Number)',
+                        // 16 -> 'IS ONE OF (Sensitive)',
+                        // 17 -> 'IS NOT ONE OF (Sensitive)'
+                    "c": "@example.com", // comparison value
+                    "v": true, // value served when the rule is evaluated
```
Comment (laliconfigcat) [inline website/docs/sdk-reference/java.md:302]: Not when evaluated but maybe when it is matching or something like this.

File: website/docs/sdk-reference/java.md [hunk 8]
```diff
@@ -246,26 +378,65 @@ ConfigCatClient client = ConfigCatClient.newBuilder()
...
+                .readTimeout(2, TimeUnit.SECONDS) // set the read timeout to 2 seconds
+                .build())
+    .build("<PLACE-YOUR-SDK-KEY-HERE>");
+```
+OkHttpClient's default timeout is 10 seconds.
+
 > As the ConfigCatClient SDK maintains the whole lifetime of the internal http client, it's being closed simultaneously with the ConfigCatClient, refrain from closing the http client manually.
 
-### Force refresh
+## Force Refresh
```
Comment (laliconfigcat) [inline website/docs/sdk-reference/java.md:395]:
```suggestion
## Force refresh
```

File: website/docs/sdk-reference/java.md [hunk 9]
```diff
@@ -246,26 +378,65 @@ ConfigCatClient client = ConfigCatClient.newBuilder()
...
+ConfigCatClient client = ConfigCatClient.newBuilder()
+    .logLevel(LogLevel.INFO)
+    .build("<PLACE-YOUR-SDK-KEY-HERE>");
+```
+
+Available log levels:
+
+| Level      | Description                                                                             |
+| ---------- | --------------------------------------------------------------------------------------- |
+| `NO_LOG`   | Turn the ConfigCat logging off.                                                         |
```
Comment (laliconfigcat) [inline website/docs/sdk-reference/java.md:417]:
```suggestion
| `NO_LOG`   | Turn the logging off.                                                         |
```

**Review by laliconfigcat — CHANGES_REQUESTED** [review]

File: website/docs/sdk-reference/java.md (hunk 5 above)
Comment (sigewuzhere) [reply website/docs/sdk-reference/java.md:253]: nekem tetszik h minden benne van

File: website/docs/sdk-reference/java.md (hunk 7 above)
Comment (sigewuzhere) [reply website/docs/sdk-reference/java.md:302]: ezt a magyarazatot en sem ertem

File: website/docs/sdk-reference/java.md [hunk 10]
```diff
@@ -196,18 +204,139 @@ ConfigCatClient client = ConfigCatClient.newBuilder()
...
+                    "i": "bcfb84a7" // variation id (for analytical purposes)
+                }
+            ]
+        },
+    }
+}
+```
+> For a baseline, you can also download your configuration from ConfigCat's CDN and use it as-is from a file.
+
+If you don't need this kind of complexity then you can use the second, a much simpler (key-value) JSON format.
```
Comment (sigewuzhere) [inline website/docs/sdk-reference/java.md:312]: lehetne eze elobb es a bonyi verzio kesobb

**Review by sigewuzhere — CHANGES_REQUESTED** [review]

**Review by sigewuzhere — APPROVED** [review]

**Review by laliconfigcat — APPROVED** [review]

**Review by sigewuzhere — APPROVED** [review]

**Comment by CCatAti** [comment]
Test Ok on TEST env.
Included links are working!

**Pull request merged**